dao/src/main/java/org/thingsboard/server/dao/entity/BaseEntityService.java

```diff
@@ -125,6 +132,27 @@ public PageData<EntityData> findEntityDataByQuery(TenantId tenantId, CustomerId
         return this.entityQueryDao.findEntityDataByQuery(tenantId, customerId, query);
     }
 
+    @Override
+    public EntityData findEntityByTenantIdAndId(TenantId tenantId, EntityId entityId, List<String> entityFields) {
```

Comment:
This implementation is far from perfect. This is a hack, not the implementation. You should switch by entity type and use a specific service. This will allow using the cache and will have minimal overhead. I believe there is already such a method in the import/export services or somewhere else